### PR TITLE
Scc 3317

### PR DESCRIPTION
**Pull request by charmingduchess**
Adds encore=>vega redirects for cases:
- languages other than english links 
- urls for searches utilizing encore's search logic
- bib page url ending with ~$1

i had to add some error handling for the tests to pass. i think there should be rigorous qa done on the legacy/scc side as well to make sure there weren't any regressions added.

### CODE REVIEW

File: expressions.js [hunk 1]
```diff
@@ -27,9 +27,37 @@ module.exports = {
   },
   // Encore => Vega redirects
   encoreBibPage: {
-      expr: /C__Rb(\d{8})__/,
+    expr: /C__Rb(\d{8})(__|~\$1|$)/,
     handler: (match) => `${VEGA_URL}/search/card?recordId=${match[1]}`
   },
+  languagesOtherThanEnglish: {
+    expr: /C__Sf:\((a|v)%20\|%20(u|y)\)(?:.*?)l:\(?([a-z]{3})\)?/,
```
Comment (thisisstephenbetts) [inline expressions.js:34]: Looks great!

File: expressions.js [hunk 2]
```diff
@@ -27,9 +27,37 @@ module.exports = {
...
+      console.log(path)
+      try {
+      const decodedPath = decodeURIComponent(path)
+      const regEx = /(?:[(]+([^-)]+)[^)]*)/g
+      const matches = [...decodedPath.matchAll(regEx)]
+      const searchTerms = encodeURI(matches
+        .map((match) => `"${match[1]}"`).join(' '))
+        return searchTerms
+      } catch (e) {
+        // this is here because one of the tests was causing a malformed URI error
```
Comment (thisisstephenbetts) [inline expressions.js:52]: This is interesting — do you know where that happens? Is it from the `decodeURIComponent` call? If you could add the problematic URI as a reply here I'd like to take a look — probably I fat-thumbed the URI.

File: expressions.js [hunk 3]
```diff
@@ -27,9 +27,37 @@ module.exports = {
   },
   // Encore => Vega redirects
   encoreBibPage: {
-      expr: /C__Rb(\d{8})__/,
+    expr: /C__Rb(\d{8})(__|~\$1|$)/,
```
Comment (thisisstephenbetts) [inline expressions.js:30]: This looks good, but if you want to simplify a bit I _think_ you could probably just reduce this to `expr: /C__Rb(\d{8})/`. Yours is more precise for sure, so it's a toss up which is better.

**Review by thisisstephenbetts — APPROVED** [review]
Excellent stuff Vera! Thank you

I made one minor suggestion, but it isn't clear to me which is better — yours is more precise. I also asked a question about the failing URI but that's just for the PR, not code changes.

File: expressions.js (hunk 2 above)
Comment (charmingduchess) [reply expressions.js]: Ooh I got it. l realized it was having an issue with the percent sign. Removed and now there is no error. I still think that the 404 is not sensitive enough though.

Comment (charmingduchess) [reply expressions.js]: /record=&%!^/ was the problematic path

File: expressions.js (hunk 3 above)
Comment (charmingduchess) [reply expressions.js:30]: i don't mind the extra characters, i think it documents what cases we're expecting.

Comment (thisisstephenbetts) [reply expressions.js:30]: 👍🏽

File: expressions.js (hunk 2 above)
Comment (thisisstephenbetts) [reply expressions.js]: Good catch. Makes sense that's a bad URL

**Pull request merged**